lark/parsers/lalr_analysis.py

```diff
@@ -279,9 +282,97 @@ def compute_lalr1_states(self):
         else:
             self.parse_table = IntParseTable.from_ParseTable(_parse_table)
 
+    # Generation of conflict resolution hints
+    def __build_reduce_reduce_error(self, la, rules):
+        """Build a GrammarError with conflict resolution hints."""
+        msg = "Reduce/Reduce collision in %s between the following rules:" % la
+        msg += "".join("\n\t\t- %s" % rule for rule in rules)
+
+        common_ancestors = self.__find_closest_common_ancestors(rules)
+        msg += "\n\nThe conflict may come from"
+
+        for rules, ancestors in common_ancestors.items():
+            a = ' or '.join("<%s>" % ancestor.name for ancestor in ancestors)
+            rule1, rule2 = sorted(rule.name for rule in rules)
+            msg += "\n\t\t- <%s> and <%s>: %s" % (rule1, rule2, a)
+
+        return GrammarError(msg)
+
+    def __find_closest_common_ancestors(self, conflicting_rules):
```

Comment:
Again, I would prefer to not use `__dunder` names

Reply:
Fixed as well